src/tree/updater_gpu_hist.cu

```diff
@@ -269,7 +270,13 @@ struct GPUHistMakerDevice {
     std::fill(node_sum_gradients.begin(), node_sum_gradients.end(),
               GradientPair());
 
-    auto sample = sampler->Sample(dh_gpair->DeviceSpan(), dmat);
+    if (d_gpair.size() != dh_gpair->Size()) {
+      d_gpair.resize(dh_gpair->Size());
+    }
+    thrust::copy(thrust::device, dh_gpair->ConstDevicePointer(),
+                 dh_gpair->ConstDevicePointer() + dh_gpair->Size(),
+                 d_gpair.begin());
+    auto sample = sampler->Sample(dh::ToSpan(d_gpair), dmat);
```

Comment:
We have to make a copy here since `Sample()` is expected to modify the incoming `gpair` parameter? Mutating gpairs would be bad when we fit multiple trees in a single round.

Reply:
> modify the incoming gpair parameter?

Yes.

Reply:
Thanks guys!

Reply:
Fix confirmed. Thanks.